### PR TITLE
Feature/login

### PR DESCRIPTION
**Pull request by santiagoRD**
## Summary

Create Login screen, add validations and integration with service and create tests

## Screenshots

Captura de Pantalla 2021-02-25 a la(s) 10 20 25 a  m

## Trello Card

https://trello.com/c/DvNtnkGz/80-login

### CODE REVIEW

File: src/components/Input/index.tsx [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
 interface Props extends InputHTMLAttributes<HTMLInputElement> {
-  inputRef?: React.Ref<HTMLInputElement>;
+  inputRef: React.Ref<HTMLInputElement>;
   label?: string;
   errorMessage?: string;
+  i18nKey: string;
 }
 
-function InputText({ inputRef, label, errorMessage, name, ...rest }: Props) {
+function InputText({ inputRef, label, errorMessage, name, i18nKey, ...rest }: Props) {
```
Comment (manugo-dev) [inline src/components/Input/index.tsx]: This prop seems strange to me, I would prefer that you send it directly in the "name" prop

File: src/components/Input/index.tsx [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
       {label && (
         <label className={`m-bottom-1 ${styles.label}`} htmlFor={name}>
-          {i18next.t(`${I18N_CONFIG.key}:${name}`)}
+          {i18next.t(`${i18nKey}:${name}`)}
         </label>
       )}
       <input className={styles.input} ref={inputRef} id={name} name={name} {...rest} />
       <span role="alert" className={styles.error}>
-        {errorMessage ? i18next.t(`${I18N_CONFIG.key}:${errorMessage}`) : ''}
+        {errorMessage ? i18next.t(`${i18nKey}:${errorMessage}`) : ''}
```
Comment (manugo-dev) [inline src/components/Input/index.tsx]: same for it

File: src/screens/Login/index.tsx [hunk 3]
```diff
@@ -1,10 +1,91 @@
...
+  const { register, errors, handleSubmit } = useForm<LoginUser>({
+    mode: 'onChange',
+    reValidateMode: 'onBlur'
+  });
+  const [, loading, error, sendRequest] = useLazyRequest({
+    request: UserService.loginUser,
+    withPostSuccess: (data, headers) => {
+      setSuccess(`${I18N_CONFIG.key.login}:${SUCCESS_MESSAGES.userLogged}`);
+      // eslint-disable-next-line no-console
+      console.log(data, headers);
```
Comment (manugo-dev) [inline src/screens/Login/index.tsx]: Delete this

File: src/screens/Login/index.tsx [hunk 4]
```diff
@@ -1,10 +1,91 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router';
+import { useForm, SubmitHandler } from 'react-hook-form';
 import i18next from 'i18next';
 
-import { I18N_CONFIG, SIGNUP_BUTTONS } from 'constants/index';
+import { ERROR_MESSAGES, I18N_CONFIG, LOGIN_FIELDS, SIGNUP_BUTTONS, SUCCESS_MESSAGES } from 'constants/index';
```
Comment (manugo-dev) [inline src/screens/Login/index.tsx]: check if you can import without index file declaration: ... from 'constants'

File: src/screens/Login/login.test.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import userEvent from '@testing-library/user-event';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import { createMemoryHistory } from 'history';
+
+import { baseURL } from 'config/api';
+import { RESPONSE_STATUS } from 'constants/index';
+import { mockBadRequest } from 'mocks';
+
+import Login from '.';
```
Comment (manugo-dev) [inline src/screens/Login/login.test.tsx]: `import Login from  './index';`

File: src/screens/Login/index.tsx [hunk 6]
```diff
@@ -1,10 +1,91 @@
...
+          type="submit"
+          aria-label="login"
+        >
+          {i18next.t(`${I18N_CONFIG.key.login}:${SIGNUP_BUTTONS.login}`)}
+        </button>
+        <button
+          className={`${styles.formButton} ${styles.secondaryButton} full-width`}
+          type="button"
+          aria-label="signup"
+          onClick={() => history.push('/sign_up')}
```
Comment (manugo-dev) [inline src/screens/Login/index.tsx]: `history.push(paths.signUp)`

File: src/screens/SignUp/index.tsx [hunk 7]
```diff
@@ -101,14 +109,15 @@ function SignUp() {
...
         >
-          {i18next.t(`${I18N_CONFIG.key}:${SIGNUP_BUTTONS.signUp}`)}
+          {i18next.t(`${I18N_CONFIG.key.signup}:${SIGNUP_BUTTONS.signUp}`)}
         </button>
         <button
-          className={`${styles.formButton} ${styles.loginButton} full-width`}
+          className={`${styles.formButton} ${styles.secondaryButton} full-width`}
           type="button"
           aria-label="login"
+          onClick={() => history.push('/')}
```
Comment (manugo-dev) [inline src/screens/SignUp/index.tsx]: `history.push(routes.home)`

File: src/screens/SignUp/signUp.test.tsx [hunk 8]
```diff
@@ -1,25 +1,38 @@
...
+);
 beforeAll(() => server.listen());
 afterEach(() => server.resetHandlers());
 afterAll(() => {
   server.close();
   cleanup();
 });
 describe('Testing SignUp Component', () => {
+  const history = createMemoryHistory();
+  history.push('/sign_up');
```
Comment (manugo-dev) [inline src/screens/SignUp/signUp.test.tsx]: Use of routes constants

File: src/screens/SignUp/signUp.test.tsx [hunk 9]
```diff
@@ -1,25 +1,38 @@
...
   beforeEach(() => {
-    render(<SignUp />);
+    render(
+      <Router history={history}>
+        <Route path="/sign_up">
```
Comment (manugo-dev) [inline src/screens/SignUp/signUp.test.tsx]: Use of routes constants

**Review by manugo-dev — CHANGES_REQUESTED** [review]

File: src/config/api.js [hunk 10]
```diff
@@ -1,7 +1,7 @@
 import { create } from 'apisauce';
 import { CamelcaseSerializer, SnakecaseSerializer } from 'cerealizr';
 
-const baseURL = process.env.REACT_APP_BASE_URL;
+export const baseURL = process.env.REACT_APP_BASE_URL || '';
```
Comment (CrisVega08) [inline src/config/api.js]: Why do you need export baseURL if you can access with process.env?

File: src/hooks/useRequest.ts [hunk 11]
```diff
@@ -1,36 +1,36 @@
...
+import { ApiErrorResponse, ApiOkResponse, HEADERS, PROBLEM_CODE, ApiResponse } from 'apisauce';
 
 import { Nullable } from 'utils/types';
 
-import { useMountedRef } from './useMountedRef';
-
 export type Error<E> = { problem: PROBLEM_CODE; errorData?: E };
 type Request<P, D, E> = (params: P) => Promise<ApiResponse<D, E>>;
-type Success<D> = (data?: D) => void;
+type Success<D, HEADERS> = (data?: D, headers?: HEADERS) => void;
```
Comment (CrisVega08) [inline src/hooks/useRequest.ts]:
```suggestion
type Success<D, H> = (data?: D, headers?: H) => void;
```

File: src/hooks/useRequest.ts [hunk 12]
```diff
@@ -42,27 +42,33 @@ const executeAsyncRequest = async <P, D, E>({
   onPrefetch();
   const response = await request(values);
```
Comment (CrisVega08) [inline src/hooks/useRequest.ts:43]:
```suggestion
  const { ok, data, headers, problem } = await request(values);
```

File: src/screens/Login/index.tsx (hunk 4 above)
Comment (CrisVega08) [inline src/screens/Login/index.tsx]:
```suggestion
import { ERROR_MESSAGES, I18N_CONFIG, LOGIN_FIELDS, SIGNUP_BUTTONS, SUCCESS_MESSAGES } from 'constants';
```

File: src/screens/Login/login.test.tsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+const server = setupServer();
+rest.post(`${baseURL}/users/sign_in`, (_, res, ctx) =>
```
Comment (CrisVega08) [inline src/screens/Login/login.test.tsx]:
```suggestion
rest.post('/users/sign_in', (_, res, ctx) =>
```

File: src/screens/SignUp/index.tsx [hunk 14]
```diff
@@ -20,17 +21,19 @@ import wLogo from 'assets/wLogo.png';
...
     reValidateMode: 'onBlur'
   });
   const password = useRef({});
   password.current = watch('password', '');
-  const [state, loading, error, sendRequest] = useLazyRequest({
+  const [, loading, error, sendRequest] = useLazyRequest({
     request: UserService.createUser,
     withPostSuccess: () => {
       setSuccess(`${I18N_CONFIG.key}:${SUCCESS_MESSAGES.userCreated}`);
+      history.push('/');
```
Comment (CrisVega08) [inline src/screens/SignUp/index.tsx]: use constante Routes

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

**Review by CrisVega08 — APPROVED** [review]

**Review by manugo-dev — APPROVED** [review]

**Pull request merged**